src/components/buttons/ProgressButton.js

```diff
@@ -1,21 +1,36 @@
-import React, { useState } from 'react'
-import { View, StyleSheet } from 'react-native'
+import React, { useState, useEffect, useRef } from 'react'
+import { View, StyleSheet, Animated } from 'react-native'
```

Comment:
If this works there is not need to change it now, but `Animated` from `react-native` is not the best way to do animations, as it requires communication between the JS thread and the native one. If the JS thread is busy, a frame drop can be seen in the animation. Use `useNativeDriver: true` option for now.

`Animated` from `react-native-animatable` or
`Animated` from `react-native-reanimated`

Are preferred options.

Reply:
Property useNativeDriver cant be set to true in this case. Native Driver doesn't support width property.